subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -571,6 +572,9 @@ private void calculateLauncherJvmArgs(GradleInvocation gradleInvocation) {
         buildJvmOpts.add("-ea");
 
         if (isDebug()) {
+            if (System.getenv().containsKey("CI")) {
```

Comment:
I wonder if we should go ahead and create a `RuntimeEnvironmentDetector` or some such that encapsulates the "am I on CI?" mechanics.  That might also help with testing.

Reply:
Maybe something as a `TestPrecondition`?

Reply:
That would work, or just have a way to force it to consider the current environment as CI (e.g. `executor.runtimeEnvironmentDetector.forceCIEnvironment()`) so you could at least validate that "when the detector thinks I'm on CI, then I get this error".  

Stepping back, does `executer.withEnvironmentVars()` not work for testing this?

Reply:
Oh, nvm, this is when you are configuring the executer, not when you're running it.  🙄